app/src/main/java/com/asma/tasky/feature_management/domain/task/use_case/AddTaskUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.asma.tasky.feature_management.domain.task.use_case
+
+import com.asma.tasky.R
+import com.asma.tasky.core.util.Resource
+import com.asma.tasky.core.util.UiText
+import com.asma.tasky.feature_management.domain.AgendaItem
+import com.asma.tasky.feature_management.domain.task.repository.TaskRepository
+import javax.inject.Inject
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class AddTaskUseCase @Inject constructor(
+    private val repository: TaskRepository
+) {
+
+    operator fun invoke(task: AgendaItem.Task): Flow<Resource<Unit>> = flow {
+        if (task.title.isBlank()) {
+            emit(Resource.Error(message = UiText.StringResource(R.string.invalid_task)))
+            return@flow
+        }
+        repository.addTask(task)
+        emit(Resource.Success(Unit))
+    }
```

Comment:
There's no need to return a flow if your function has only one return value. Then simply make it a suspend function.